app/api_v1/resources.py

```diff
@@ -0,0 +1,369 @@
+                'limit=' + str(limit) + '&page=' + str(page_no - 1)
+        else:
+            previous_page = 'None'
+
+        bucketlists = bucketlists_per_page.items
+
+        response = {'bucketlists': marshal(bucketlists, bucketlist_serializer),
+                    'has_next': has_next,
+                    'pages': all_bucketlists,
+                    'previous_page': previous_page,
+                    'next_page': next_page
+                    }
+        return response
+
+
+class BucketlistApi(Resource):
+
+    @auth.login_required
+    def get(self, id):
+        """
+        Returns the bucketlist with the given id
+
+        parameters:
+          -id
+
+        """
+
+        got_list = Bucketlist.query.filter_by(
+            id=id, created_by=g.user.id).first()
+        if not got_list == None:
+            return marshal(got_list, bucketlist_serializer)
+        else:
+            return {'message': 'Specified bucketlist not found.'}, 404
+
+    @auth.login_required
+    def put(self, id):
+        """
+        Edits/updates the bucketlist with the given id
+
+        parameters:
+          -name
+          -id
+        """
+        parser = reqparse.RequestParser()
+        parser.add_argument('name', type=str, required=True,
+                            help='New name not provided', location='form')
+        args = parser.parse_args()
+        new_name = args['name']
+
+        got_list = Bucketlist.query.filter_by(
+            id=id, created_by=g.user.id).first()
+        if not got_list == None:
+            got_list.name = new_name
+            db.session.add(got_list)
+            db.session.commit()
+            return {"successfully edited ": marshal(got_list, bucketlist_serializer)}, 200
+
+        else:
+            return {'message': 'Specified bucketlist not found.'}, 404
+
+    @auth.login_required
+    def delete(self, id):
+        """
+        Deletes the bucketlist with the given id
+
+        parameters:
+          -id
+        """
+        to_delete = Bucketlist.query.filter_by(
+            created_by=g.user.id, id=id).first()
+        if not to_delete == None:
+            db.session.delete(to_delete)
+            db.session.commit()
+            return {'message': 'Bucketlist successfully deleted'}, 200
+        else:
+            return {'message': 'Not deleted, Bucketlist does not exist'}, 404
+
+
+class BucketlistItemCreateApi(Resource):
+    def __init__(self):
+        self.parse = reqparse.RequestParser()
+        self.parse.add_argument('name', type=str, required=True,
+                                help='Item name not provided',
+                                location='form')
+        self.parse.add_argument('description', type=str,
+                                location='form')
+
+    @auth.login_required
+    def post(self, id):
+        """
+        Creates a new item in the bucketlist with the provided id
+
+        parameters:
+          -name
+          -description
+          -id
```

Comment:
It would be better if the id was auto created in the db maybe as an auto incrementing primary key